src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/StartupProjectRegistrar.cs

```diff
@@ -87,15 +80,20 @@ internal async Task OnProjectChangedAsync(IProjectVersionedValue<IProjectSubscri
             bool isDebuggable = await _launchProviders.Value.IsDebuggableAsync()
                                                             .ConfigureAwait(false);
 
+            IVsStartupProjectsListService startupProjectsListService = await _startupProjectsListService.GetValueAsync()
+                                                                                                        .ConfigureAwait(true);
```

Comment:
Why does this need to return to the UI thread?

Reply:
Only the cast runs on the UI thread, package init, service creation will occur on a background thread if possible.